avogadro/rendering/solid_first_fs.glsl

```diff
@@ -66,6 +68,47 @@ float lerp(float a, float b, float f)
     return a + f * (b - a);
 }
 
+float rand(vec2 co){
+  return fract(sin(dot(co.xy ,vec2(12.9898,78.233))) * 43758.5453);
+}
+
+float depthToZ(float depth) {
+  float eyeZ = ((height * 0.57735) /2.0);
+  float near = 5.0;
+  float far = 600.0;
+  float depthNormalized = 2.0 * depth - 1.0;
+  return 2.0 * near * far / (far + near - depthNormalized * (far - near));
+}
+
+float calcBlur(float z, float pixelScale) {
+  return clamp(abs(z - 39.0), 0.0, 0.5*pixelScale);
+}
+
+vec4 applyBlur(vec2 texCoord) {
+  float pixelScale = max(width, height);
+  float origZ = depthToZ(texture2D(inDepthTex, texCoord).x);
+  float blurAmt = calcBlur(origZ, pixelScale);
+  
+  float total = 1.0;
+  vec4 color = texture2D(inRGBTex, texCoord);
+  // number of samples can be optimized.
+  for (int i = 0; i < 64; i++) {
+    float t = (float(i) / float(64));
+    float angle = (t * 4.0) * 6.28319; 
+    float radius = (t * 2. - 1.); 
+    angle += 1.0 * rand(gl_FragCoord.xy);
+    vec2 offset = (vec2(cos(angle), sin(angle)) * radius * 0.05 * blurAmt) / pixelScale;
+    float z = depthToZ(texture2D(inDepthTex, texCoord + offset).x);
+    float sampleBlur = calcBlur(z, pixelScale);
+    float weight = float((z >= origZ) || (sampleBlur >= blurAmt * radius + 0.));
```

Comment:
float weight = 1.0 - smoothstep(0.0, 1.0, abs(z - origZ) / blurAmt); 
// using smoothstep to soften weight transitions
This modification uses smoothstep to create a more gradual transition in weights applied to sampled colors, potentially reducing artifacts like banding. Reviews @ghutchis

Reply:
I think the smoothstep() would be nice here, since it's also easier to read.